### PR TITLE
Initscripts workarounds

### PR DESCRIPTION
**Pull request by tyll**
Add two fixes/workarounds in case network.service from initscripts has issues. See commit messages for details.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -23,11 +23,26 @@
...
       enabled: yes
+  when:
+    - network_provider == "nm"
+
+- name: Enable network service
+  service:
+      name: "{{ network_service_name }}"
+      enabled: yes
+  when:
+    -  network_provider == "initscripts"
```
Comment (pcahyna) [inline tasks/main.yml]: When will it be actually started?

Comment (pcahyna) [reply tasks/main.yml]: I mean, don't you still need to start it to apply the new settings?

Comment (tyll) [reply tasks/main.yml]: No, the new settings are applied by the module (it issues `ifup` directly). The service only ensures that the profiles on disk are applied when the system starts (or when the service is started).

**Comment by tyll** [comment]
Thank you for the fast review!

**Comment by pcahyna** [comment]
Will it (the network service) be really started after boot, if there are network profiles for unknown devices? (Aren't we just postponing the problem to a later moment?)

**Comment by tyll** [comment]
> Will it (the network service) be really started after boot, if there are network profiles for unknown devices? (Aren't we just postponing the problem to a later moment?)

With the workaround users can specify `peristent_state: absent` as a profile to remove all unmentioned profiles to make it more likely that this will not be a problem.

Also even if the network service failed to start from systemd perspective, it will still try to bring up all the profiles. So the system will still have the network access that is specified in the profiles.

**Pull request merged**